pkg/generators/forwarding/fluentd/output_conf_legacy_test.go

```diff
@@ -0,0 +1,1376 @@
+              static_index_name app-write
+            </elasticsearch_index_name>
+          </filter>
+
+          <filter **>
+            @type elasticsearch_genid_ext
+            hash_id_key viaq_msg_id
+            alt_key kubernetes.event.metadata.uid
+            alt_tags "#{ENV['GENID_ALT_TAG'] || 'kubernetes.var.log.containers.logging-eventrouter-*.** kubernetes.var.log.containers.eventrouter-*.** kubernetes.var.log.containers.cluster-logging-eventrouter-*.** kubernetes.journal.container._default_.kubernetes-event'}"
+          </filter>
+
+          #flatten labels to prevent field explosion in ES
+          <filter ** >
+            @type record_transformer
+            enable_ruby true
+            <record>
+              kubernetes ${!record['kubernetes'].nil? ? record['kubernetes'].merge({"flat_labels": (record['kubernetes']['labels']||{}).map{|k,v| "#{k}=#{v}"}}) : {} }
+            </record>
+            remove_keys $.kubernetes.labels
+          </filter>
+
+          # Relabel specific source tags to specific intermediary labels for copy processing
+
+          <match **_default_** **_kube-*_** **_openshift-*_** **_openshift_** journal.** system.var.log**>
+            @type relabel
+            @label @_INFRASTRUCTURE
+          </match>
+
+          <match kubernetes.**>
+            @type relabel
+            @label @_APPLICATION
+          </match>
+
+          <match linux-audit.log** k8s-audit.log** openshift-audit.log**>
+            @type relabel
+            @label @_AUDIT
+          </match>
+
+          <match **>
+            @type stdout
+          </match>
+
+        </label>
+
+        # Relabel specific sources (e.g. logs.apps) to multiple pipelines
+        <label @_APPLICATION>
+          <match **>
+            @type copy
+
+
+
+            <store>
+              @type relabel
+              @label @_LEGACY_SYSLOG
+            </store>
+          </match>
+        </label>
+        <label @_AUDIT>
+          <match **>
+            @type copy
+
+
+
+            <store>
+              @type relabel
+              @label @_LEGACY_SYSLOG
+            </store>
+          </match>
+        </label>
+        <label @_INFRASTRUCTURE>
+          <match **>
+            @type copy
+
+
+
+            <store>
+              @type relabel
+              @label @_LEGACY_SYSLOG
+            </store>
+          </match>
+        </label>
+
+        # Relabel specific pipelines to multiple, outputs (e.g. ES, kafka stores)
+
+        # Ship logs to specific outputs
+
+
+        <label @_LEGACY_SYSLOG>
+          <match **>
+            @type copy
+            #include legacy Syslog
+            @include /etc/fluent/configs.d/syslog/syslog.conf
+          </match>
+        </label>
+      `
+
+      results, err := generator.Generate(forwarderSpec)
+      Expect(err).Should(Succeed())
+      Expect(results).To(EqualTrimLines(legacyConf))
+    })
+  })
+
+  Context("based on legacy syslog and fluentd forward method", func() {
```

Comment:
can we keep just this one test case, other two are included in this one it seems

Reply:
The test cases are distinct from each other. They cover all the if-conditionals in the template code, a) only fluent, b) only syslog, c) both. The configuration looks different for each three and it should be tested in isolation each time.